zio-cli/shared/src/main/scala/zio/cli/HelpDoc.scala

```diff
@@ -24,19 +24,28 @@ sealed trait HelpDoc { self =>
 
   def |(that: HelpDoc): HelpDoc = if (self.isEmpty) that else self
 
-  def isHeader: Boolean =
+  def isEmpty: Boolean =
     self match {
       case HelpDoc.Empty                 => true
-      case HelpDoc.DescriptionList(xs)   => true
-      case HelpDoc.Sequence(left, right) => left.isDescriptionList
-      case HelpDoc.Enumeration(xs)       => false
+      case HelpDoc.DescriptionList(xs)   => xs.forall(_._2.isEmpty)
+      case HelpDoc.Sequence(left, right) => left.isEmpty && right.isEmpty
+      case HelpDoc.Enumeration(xs)       => xs.forall(_.isEmpty)
       case _                             => false
     }
 
-  def isParagraph: Boolean =
+  def isHeader: Boolean =
     self match {
       case HelpDoc.Header(_, _)          => true
+      case HelpDoc.Sequence(left, right) => left.isHeader
+      case HelpDoc.Enumeration(xs)       => xs.head.isHeader
+      case _                             => false
+    }
+
+  def isParagraph: Boolean =
+    self match {
+      case HelpDoc.Paragraph(_)          => true
       case HelpDoc.Sequence(left, right) => left.isParagraph
+      case HelpDoc.Enumeration(xs)       => xs.head.isParagraph
```

Comment:
I would delete this line:
 
```suggestion
```

Reply:
An enumeration is a bullet point list, like this:

 - foo
 - bar 
 - baz

Whereas `Sequence` is a "virtual" element that merely allows vertically stacking blocks on top of each other.